### ISSUE TITLE
Proposal: Don't use averages to smooth out throttle

### ISSUE THREAD

**Issue by kscholty** [digimer/Mermaids-Rest#13]
This is a suggestion. 
Instead of calculating an average you could use a simple control loop like

```
changeValue = currentDACValue-newDACValue
if ( abs(changeValue) > maxAllowedDelta) {adjust newDACValue by +/-max delta}
currentDACValue = newtDACValue
```

To make sure that the values don't change too radically. This is simpler and more linear than the average method.

**Comment by jokke-ilujo** [comment]
I agree with the topic of this issue, and would advice to simplify even more. Just don't try to pretend being the motor controller before the motor controller. The throttle is to indicate to the controller where the helm wants to end up to. It's the controller's calibration and job to get there. That's why the controller has for example the hall sensor signals coming to it, not to the throttle.

Lets take the current loop configuration as an example:
We have delay time of 100ms and we set the max delta to 100 steps (digits sent to the DAC).
On emergency situation if you yank the throttle from full forward to full reverse, it takes about 2 seconds before the engine goes even to neutral providing you forwards thrust still and further 2 second before the controller receives the info that you actually wanted full reverse thrust. While in ICE world this might be expected, with electrical propulsion there is no excuse for that. The motor controller is much better equipped to handle the electrically breaking the motor to stop and then reversing it.

Also it's way too easy to end up to a situation where for example the motor never gets out of neutral if the neutral start <-> neutral center <-> neutral stop range is larger than the max delta. (Especially if you swap controllers. As the neutral zone might be mutable like you indicated in the original code.)
